hoodie-utilities/src/main/java/com/uber/hoodie/utilities/deltastreamer/DeltaSync.java

```diff
@@ -453,14 +453,16 @@ private HoodieWriteConfig getHoodieClientConfig(SchemaProvider schemaProvider) {
     HoodieWriteConfig.Builder builder =
         HoodieWriteConfig.newBuilder()
             .withProps(props)
-            .withPath(cfg.targetBasePath)
-            .combineInput(cfg.filterDupes, true)
             .withCompactionConfig(HoodieCompactionConfig.newBuilder()
+                .fromProperties(HoodieIndexConfig.newBuilder()
```

Comment:
Would n't `withProps(props)` add it in?  because even when the index config is built using `            .withIndexConfig(HoodieIndexConfig.newBuilder().withIndexType(HoodieIndex.IndexType.BLOOM).build())	` it adds to underlying props anyway

```
public Builder withIndexConfig(HoodieIndexConfig indexConfig) {
      props.putAll(indexConfig.getProps());
      isIndexConfigSet = true;
      return this;
    }
```

and all the config getters query `HoodieWriteConfig` props ?

Reply:
withProps(props) do add it, but withCompactionConfig would overwrite it with the default value.
for example, add 'hoodie.clean.automatic=false' with  withProps(props), if it was not added to 
HoodieCompactionConfig, the HoodieCompactionConfig build method would set default value:
```
 setDefaultOnCondition(props, !props.containsKey(AUTO_CLEAN_PROP), AUTO_CLEAN_PROP,
          DEFAULT_AUTO_CLEAN);
```
and 
```
public Builder withCompactionConfig(HoodieCompactionConfig compactionConfig) {
      props.putAll(compactionConfig.getProps());
      isCompactionConfigSet = true;
      return this;
    }
```
would overwrite the  props with that  default value
so the HoodieWriteConfig would have 'hoodie.clean.automatic=true',  that is the default value

Reply:
Original intention of this block here, is to actually override a few things consistently from the delta streamer command line args.. You are right, it does not work like intended to. 

Alternatively, what do you think about simply moving `withProps(props)` to the end.. i.e user could override the command line arg if they needed to (and know what they are doing).. I feel thats a simpler usage than wiring in `fromProps(..)` in multiple places..  lmk

Reply:
yes, It's a litter weird to write fromProps(..) in multiple  places.
If move withProps(props) to the end (or after withCompactionConfig), the props would override the command line args before this method. 
Users need to be careful not to put these parameters will appear on the command line in the configuration file, otherwise the results will make them feel a bit strange.

Reply:
yeah. that probably the confusion I had before as well :) ..  Its atleast simpler to explain instead of saying "oh this prop is overridden, this is not" 

as long as we document it, it should be fine? https://github.com/apache/incubator-hudi/tree/asf-site